// abm-components/src/components/abmmodal/abmmodal-style.js
import { css } from 'lit';

export const styles = css`
.principal-div {
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative; /* Necesario si usas elementos posicionados dentro */
  background-color: rgba(0, 0, 255, 0.5);

}

.modal {
  width: 50%;
  height: 50%;
  background-color: rgba(173, 216, 230, 0.5); /* Azul claro con 80% de opacidad */
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  border: 6px solid yellow;
  border-radius: 30px;
  z-index: 1; /* Asegúrate de que el modal tenga un z-index menor que el banner */
}


  .buttom{
    position: absolute;
    top: 10%;
    right: 5%;
    border-radius: 30px;
    width: 7%;
    height: 8%;
    display:flex;
    justify-content: center;
    align-items: center;
    background-color: #007bff; /* Azul */
    border: 2px solid #0056b3; /* Azul más oscuro */
    color: white; /* Texto blanco */
    font-size: 12px;
   }

   .title {
  display: flex;
  justify-content: center;
  align-items: center; /* Asegura que el texto esté centrado verticalmente */
  font-size: 30px;
  margin-top: 5%;
  max-width: 80%; /* Ajusta el ancho máximo según sea necesario */
  word-wrap: break-word; /* Permite que el texto se ajuste a la siguiente línea */
  text-align: center; /* Centra el texto horizontalmente */
}

.banner {
  width: 30%;
  height: 30vh;
  border: 2px solid #0056b3; /* Azul más oscuro */
  position: absolute; /* Necesario para poder posicionarlo sobre el modal */
  top: 32%; /* Ajusta la posición vertical según sea necesario */
  left: 50%; /* Centra horizontalmente el banner */
  transform: translateX(-50%); /* Centra el banner horizontalmente */
  z-index: 2; /* Asegura que el banner esté por encima del modal */
  text-align: center;
  border: 6px solid white;
  border-radius: 30px;
  font-size: 20px;
}

.banner-buttom-right {
  position: absolute; /* Cambia a absolute */
  bottom: 5%; /* Ajusta según sea necesario para el espacio desde el fondo */
  right: 5%; /* Ajusta según sea necesario para el espacio desde la derecha */
  border-radius: 30px;
  width: 15%; /* Puedes ajustar el ancho como prefieras */
  height: 15%; /* Ajusta la altura según sea necesario */
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #007bff; /* Azul */
  border: 2px solid #0056b3; /* Azul más oscuro */
  color: white; /* Texto blanco */
  font-size: 12px;
}

.banner-buttom-left{
  position: absolute; /* Cambia a absolute */
  bottom: 5%; /* Ajusta según sea necesario para el espacio desde el fondo */
  right: 22%; /* Ajusta según sea necesario para el espacio desde la derecha */
  border-radius: 30px;
  width: 15%; /* Puedes ajustar el ancho como prefieras */
  height: 15%; /* Ajusta la altura según sea necesario */
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #007bff; /* Azul */
  border: 2px solid #0056b3; /* Azul más oscuro */
  color: white; /* Texto blanco */
  font-size: 12px;
}

   @media (max-width: 768px) {
    
   .title{
    display: flex;
    justify-content: center;
    font-size: 20px;
    margin-top: 5%;

   }

    .buttom{
    position: absolute;
    top: 2%;
    right: 5%;
    border-radius: 30px;
    width: 10%;
    height: 8%;
    display:flex;
    justify-content: center;
    align-items: center;
    background-color: #007bff; /* Azul */
    border: 2px solid #0056b3; /* Azul más oscuro */
    color: white; /* Texto blanco */
    font-size: 12px;
   }
}


`;
