// abm-components/src/components/screen-maintenance/screen-maintenance-styles.js
import { css } from 'lit';


export const styles = css`
.screen-director {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100vh;
  background-color: #1c7dc9;
}
.central-square {
  display: flex;
  flex-direction: column; /* Mantiene header y body en columna */
  justify-content: flex-start; /* Alinea el contenido en la parte superior */
  border: 4px solid #fff;
  border-radius: 7%;
  width: 50%;
  height: 50vh;
}
.header {
  display: flex;
  flex-direction: column;
  align-items: center; /* Centra el contenido en el eje horizontal */
  text-align: center; /* Centra el texto */
  padding: 5px; /* Espaciado interior reducido */
  border-bottom: 2px solid white;
}
.header-text {
  background-color: rgba(255, 255, 255, 0.8); /* Fondo blanco semitransparente */
  border-radius: 5%; /* Bordes redondeados */
  padding: 10px; /* Espaciado interior */
  max-width: 90%; /* Ancho máximo del cuadrado */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2); /* Sombra para darle profundidad */
  text-align: start;
}
.header p {
  font-weight: bold; /* Texto en negrita */
  font-size: 1.2em; /* Tamaño de fuente aumentado */
  margin: 0; /* Elimina márgenes adicionales */
}
.body {
  flex: 1; /* Ocupa el espacio restante */
  display: flex; /* Usamos flex para alinear los GIFs y el spinner */
  justify-content: space-between; /* Espacio entre los elementos */
  align-items: center; /* Centra verticalmente los elementos */
  padding: 10px; /* Espaciado interior */
  background-image: url("https://st2.depositphotos.com/1823350/10260/v/450/depositphotos_102609836-stock-illustration-computer-service-man-repairing-computer.jpg");
  background-position: center;
  background-repeat: no-repeat;
}
.body img {
  max-width: 45%; /* Asegura que las imágenes no ocupen demasiado espacio */
  height: auto; /* Mantiene la relación de aspecto */
}
.spinner {
  border: 8px solid rgba(255, 255, 255, 0.3);
  border-radius: 50%;
  border-top: 8px solid white;
  width: 50px; /* Tamaño del spinner */
  height: 50px;
  animation: spin 1s linear infinite; /* Efecto de giro continuo */
  margin: 0 20px; /* Espaciado lateral del spinner */
}
@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
.text-spinner {
  display: flex;
  flex-direction: row;
}

/* Estilos específicos para dispositivos móviles */
@media (max-width: 600px) {
  .central-square {
    padding: 3%;
    width: 80%; /* Ancho completo en móviles */
    height: 60vh; /* Aumenta la altura para hacerlo más grande */
  }
  .header p {
    font-size: 0.9em; /* Tamaño de fuente más pequeño en móviles */
  }
  .body {
    flex-direction: column; /* Cambiar a columna en móviles */
  }
  .body img {
    display: none; /* Oculta los GIFs en móviles */
  }
  .text-spinner h2 {
    font-size: 1em;
    display: flex;
    flex-direction: row;
    margin-left: 8%;
  }

  .spinner {
    width: 30px; /* Tamaño reducido del spinner */
    height: 30px;
  }
}
`;
